### PR TITLE
Tests: factor out startd_addr waiting/detection

### PR DESCRIPTION
**Pull request by rynge**
- startd_addr waiting/detection is now a reusable script baked into the container
- This solves some waiting corner cases, such as making sure we are not querying the master before it has even started up. I also added a wait for the startd log, which takes care of the gap between master and startd starting up.
- Having it as script inside just makes for cleaner code in the tests.

### CODE REVIEW

File: sbin/startd_addr.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    local condor_version
+    condor_version=$(condor_version | awk '/CondorVersion/ {print $2}')
+    python3 -c '
+import sys
+minimum = [int(x) for x in sys.argv[1].split(".")]
+maximum = [int(x) for x in sys.argv[2].split(".")]
+version = [int(x) for x in sys.argv[3].split(".")]
+sys.exit(0 if minimum <= version <= maximum else 1)
+' "$minimum" "$maximum" "$condor_version"
+}
```
Comment (brianhlin) [inline sbin/startd_addr.sh:19]: It doesn't look like this is used at all?

**Review by brianhlin — APPROVED** [review]
Thanks for the PR! One minor comment but otherwise LGTM.
